src/main/scala/Main.scala

```diff
@@ -1,3 +1,103 @@
 object Main extends App {
-  println("Hello, world!")
+  // TASK 1
+  def max(arr: Array[Int]): Option[Int] = if (arr.isEmpty) None else Some(arr.reduce((x, y) => x max y))
+
+  def min(arr: Array[Int]): Option[Int] = if (arr.isEmpty) None else Some(arr.reduce((x, y) => x min y))
+
+  def sum(arr: Array[Int]): Int = arr.fold(0)((x, y) => x + y)
+
+
+  /**
+   * Returns a card's value as an integer.
+   * Returns None if the string passed is not a valid card.
+   *
+   * @param card The card representation as string
+   * @return The card's value
+   */
+  def parse(card: String): Option[Int] = {
+    card match {
+      case "A" => Some(11)
+      case "J" | "Q" | "K" => Some(10)
+      case _ =>
+        try {
+          val number = card.toInt
+          if (number < 2 || number > 10) None else Some(number)
+        } catch {
+          case _: NumberFormatException => None
+        }
+    }
+  }
+
+  /**
+   * Takes an array of strings and calls parse for each element
+   *
+   * @param cards An array of card strings
+   * @return The values of the cards as integers
+   */
+  def parseAll(cards: Array[String]): Array[Int] = {
+    cards.map(card => parse(card).getOrElse(0)).filter(x => x != 0)
```

Comment:
You are using `Option` everywhere so far, so if you want to get rid of it you can simply use `flatten`.
```scala
def parseAll(cards: Array[String]): Array[Int] = 
  cards.map(card => parse(card)).flatten
```
or in the most concise way possible:
```scala
def parseAll(cards: Array[String]): Array[Int] = cards.flatMap(parse)
```
`flatMap` is the same as `map` followed by `flatten`. You have already made use of this in `determineHandValue`.
We will also cover this in future lectures